FoodOrdering.html

```diff
@@ -0,0 +1,172 @@
+
+    </div>
+    <div class="item drinks" data-diet="vegan">
+        <img src="./resources/images/test6.png" alt="Fruit Smoothie">
+        <h3>Fruit Smoothie</h3>
+        <p>Drinks</p>
+        <span>$5.00</span>
+        <button onclick="addToCart('Fruit Smoothie', 5.00)">Adaugă în coș</button>
+    </div>
+    <div class="item salad" data-diet="regular">
+        <img src="./resources/images/test6.png" alt="Grilled Steak">
+        <h3>Grilled Steak</h3>
+        <p>Main dishes</p>
+        <span>$18.00</span>
+        <button onclick="addToCart('Grilled Steak', 18.00)">Adaugă în coș</button>
+    </div>
+</section>
+<script>
+    const categoryFilter = document.getElementById('categoryFilter');
+    const sortFilter = document.getElementById('sortFilter');
+    const dietFilter = document.getElementById('dietFilter');
+    const priceSlider = document.getElementById('priceSlider');
+    const priceValueDisplay = document.getElementById('priceValue');
+    const products = document.querySelector('.products');
+    const items = Array.from(document.querySelectorAll('.item'));
+
+    function updateDisplay() {
+        const selectedCategory = categoryFilter.value;
+        const sortOrder = sortFilter.value;
+        const selectedDiet = dietFilter.value;
+        const maxPrice = parseFloat(priceSlider ? priceSlider.value : 100);
+
+        let visibleItems = items.filter(item => {
+            const matchesCategory = selectedCategory === 'all' || selectedCategory === 'categories' || item.classList.contains(selectedCategory);
+            const matchesDiet = selectedDiet === 'all' || item.dataset.diet === selectedDiet;
+            const priceSpan = item.querySelector('span');
+            if (!priceSpan) return false;
+            const price = parseFloat(priceSpan.textContent.replace('$', '')) || 0;
+            const matchesPrice = price <= maxPrice;
+            return matchesCategory && matchesDiet && matchesPrice;
+        });
+
+        if (sortOrder !== 'none') {
+            visibleItems = visibleItems.sort((a, b) => {
+                const nameA = a.querySelector('h3').textContent.toLowerCase();
+                const nameB = b.querySelector('h3').textContent.toLowerCase();
+                return sortOrder === 'asc' ? nameA.localeCompare(nameB) : nameB.localeCompare(nameA);
+            });
+        }
+
+        products.innerHTML = '';
+        visibleItems.forEach(item => products.appendChild(item));
+    }
+
+    categoryFilter.addEventListener('change', updateDisplay);
+    sortFilter.addEventListener('change', updateDisplay);
+    dietFilter.addEventListener('change', updateDisplay);
+    if (priceSlider) {
+        priceSlider.addEventListener('input', () => {
+            if (priceValueDisplay) {
+                priceValueDisplay.textContent = `$${parseFloat(priceSlider.value).toFixed(2)}`;
+            }
+            updateDisplay();
+        });
+    }
+    updateDisplay();
+</script>
+
+</body>
+</html>
```

Comment:
You should have a newline at the end of each line: 
https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b